### PR TITLE
OSDOCS-9735: add customize subcommand iscsi procedures

### PR DESCRIPTION
**Pull request by jldohmann**
Version(s): 4.16+

Issue: https://issues.redhat.com/browse/OSDOCS-9735

Link to docs preview:
- iBFT
   - iso: https://77322--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-bare-metal.html#installation-user-infra-machines-advanced-customizing-live-iso-iscsi-ibft_installing-bare-metal
   - pxe: https://77322--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-bare-metal.html#installation-user-infra-machines-advanced-customizing-live-pxe-iscsi-ibft_installing-bare-metal
- manual
   - iso: https://77322--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-bare-metal.html#installation-user-infra-machines-advanced-customizing-live-iso-iscsi-manual_installing-bare-metal
   - pxe: https://77322--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-bare-metal.html#installation-user-infra-machines-advanced-customizing-live-pxe-iscsi-manual_installing-bare-metal

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by jldohmann** [comment]
@jlebon here are the iSCSI procedures using the iso/pxe subcommands. PTAL and lmk what you think 😄

**Comment by jldohmann** [comment]
@HuijingHei PTAL at this docs PR, it's more iscsi stuff but using `coreos-installer iso customize` and `coreos-installer pxe customize`; i've taken the bulk of comments from #77275 and applied them here as well

File: modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+----
+<1> The script that gets run before installation. It should contain the `iscsiadm` commands for mounting the iSCSI target.
+<2> The script that gets run after installation. It should contain the command `iscsiadm --mode node --logout=all`.
+<3> The path of a single multipathed device. If there are multiple multipath devices connected, or to be explicit, you can use the World Wide Name (WWN) symlink available in `/dev/disk/by-path`.
+<4> The ignition configuration for the destination system.
+<5> The iSCSI parameter is read from the BIOS firmware.
+<6> Optional: include this parameter if you are enabling multipathing.
+endif::[]
++
+For more information about the iSCSI options supported by `dracut`, see the link:https://www.man7.org/linux/man-pages/man7/dracut.cmdline.7.html[`dracut.cmdline` manual page].
```
Comment (GroceryBoyJr) [inline modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc:65]: I love how you set this up. It's clean and easy to read. But I have a question: are the scripts you are talking about provided by us at Red Hat or do they need to be written by the user? (and, please pardon me if I missed something along the way and should have known this)

File: modules/installation-user-infra-machines-advanced-customizing-live-iscsi-manual.adoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+----
+<1> The script that gets run before installation. It should contain the `iscsiadm` commands for mounting the iSCSI target and any commands enabling multipathing.
+<2> The script that gets run after installation. It should contain the command `iscsiadm --mode node --logout=all`.
+<3> The location of the destination system. You must provide the IP address of the target portal, the associated port number, the target iSCSI node in IQN format, and the iSCSI logical unit number (LUN).
+<4> The ignition configuration for the destination system.
+<5> The iSCSI initiator, or client, name in IQN format. The initiator forms a session to connect to the iSCSI target.
+<6> The the iSCSI target, or server, name in IQN format.
+endif::[]
++
+For more information about the iSCSI options supported by `dracut`, see the link:https://www.man7.org/linux/man-pages/man7/dracut.cmdline.7.html[`dracut.cmdline` manual page].
```
Comment (GroceryBoyJr) [inline modules/installation-user-infra-machines-advanced-customizing-live-iscsi-manual.adoc:64]: same, I love the construction: it's clear and easy to read in both source and the end result.

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM, however, see questions in-line. Not errors, just conceptual questions.

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM

File: modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc (hunk 1 above)
Comment (jldohmann) [reply modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc:65]: The `dracut` commands i believe are inherited from RHEL

Comment (jldohmann) [reply modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc:65]: @GroceryBoyJr ohh i see what you mean. you're referring to the `pre-install` and `post-install` scripts mentioned. they need to be created by the user so kind of outside the scope of what we document. however, they need to include the specific things mentioned + whatever else they need for their infrastructure

File: modules/installation-user-infra-machines-advanced-customizing-live-iscsi-manual.adoc (hunk 2 above)
Comment (jldohmann) [reply modules/installation-user-infra-machines-advanced-customizing-live-iscsi-manual.adoc:64]: thanks James! 😄

File: modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+:_mod-docs-content-type: PROCEDURE
+[id="installation-user-infra-machines-advanced-customizing-live-{boot}-iscsi-ibft_{context}"]
+= Customizing a live install {boot-media} for an iSCSI boot device with iBFT
+
+You can set the iSCSI target and initiator values for automatic mounting, booting and configuration using a customized version of the live {op-system} image.
+
+.Prerequisites
+. You have an iSCSI target you want to install {op-system} on.
+. Optional: you have multipathed your iSCSI target.
```
[the remark this answers is not in the document]
Comment (jldohmann) [reply modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc:15]: @GroceryBoyJr it's got some precedence both in OCP docs and RHEL docs, so i think it's fine

File: modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    --post-install unmount-iscsi.sh \ <2>
+    --dest-device /dev/mapper/mpatha \ <3>
+    --dest-ignition config.ign \ <4>
+    --dest-karg-append rd.iscsi.firmware=1 \ <5>
+    --dest-karg-append rd.multipath=default \ <6>
+    -o custom.iso rhcos-<version>-live.x86_64.iso
+----
+<1> The script that gets run before installation. It should contain the `iscsiadm` commands for mounting the iSCSI target and any commands enabling multipathing.
+<2> The script that gets run after installation. It should contain the command `iscsiadm --mode node --logout=all`.
+<3> The path of a single multipathed device. If there are multiple multipath devices connected, or to be explicit, you can use the World Wide Name (WWN) symlink available in `/dev/disk/by-path`.
```
Comment (jlebon) [inline modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc:37]:
```suggestion
<3> The path to the device. If using multipath, the multipath device (e.g. `/dev/mapper/mpatha`). If there are multiple multipath devices connected, or to be explicit, you can use the World Wide Name (WWN) symlink available in `/dev/disk/by-path`.
```

File: modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+<4> The ignition configuration for the destination system.
```
Comment (jlebon) [inline modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc:38]:
```suggestion
<4> The Ignition configuration for the destination system.
```

File: modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc (hunk 4 above)
Comment (jlebon) [reply modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc:37]: Similarly for PXE.

**Review by jlebon — COMMENTED** [review]
Looks sane overall! Should we link to these sections from the new iSCSI sections added in the previous PR?

**Comment by kalexand-rh** [comment]
@HuijingHei, will you PTAL at this PR?

File: modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+$ coreos-installer iso customize \
+    --pre-install mount-iscsi.sh \ <1>
+    --post-install unmount-iscsi.sh \ <2>
+    --dest-device /dev/mapper/mpatha \ <3>
+    --dest-ignition config.ign \ <4>
+    --dest-karg-append rd.iscsi.firmware=1 \ <5>
+    --dest-karg-append rd.multipath=default \ <6>
+    -o custom.iso rhcos-<version>-live.x86_64.iso
+----
+<1> The script that gets run before installation. It should contain the `iscsiadm` commands for mounting the iSCSI target and any commands enabling multipathing.
```
Comment (HuijingHei) [inline modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc:35]: Overall looks good to me, one small question, not the blocker, does `mount-iscsi.sh` is like the `iscsiadm` command in https://github.com/openshift/openshift-docs/pull/77275 ?

Comment (jldohmann) [reply modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc:35]: yeah, pretty much! 😄 i think links between the two procedures is a good idea, i'll take a look at adding some post-GA

**Pull request merged**